js/modules/data.js

```diff
@@ -654,20 +656,40 @@ async function removeAllSessions(id) {
 
 // Conversation
 
+function setifyProperty(data, propertyName) {
+  const returnData = data;
+  if (returnData[propertyName]) {
+    returnData[propertyName] = new Set(returnData[propertyName]);
+  }
+  return returnData;
+}
+
+async function getSwarmNodesByPubkey(pubkey) {
+  let swarmNodes = await channels.getSwarmNodesByPubkey(pubkey);
+  if (Array.isArray(swarmNodes)) {
+    swarmNodes = new Set(swarmNodes);
+  }
+  return swarmNodes;
+}
+
 async function getConversationCount() {
   return channels.getConversationCount();
 }
 
 async function saveConversation(data) {
-  await channels.saveConversation(data);
+  const storeData = data;
+  if (storeData.swarmNodes) {
+    storeData.swarmNodes = Array.from(storeData.swarmNodes);
+  }
+  await channels.saveConversation(storeData);
 }
 
 async function saveConversations(data) {
   await channels.saveConversations(data);
 }
 
 async function getConversationById(id, { Conversation }) {
-  const data = await channels.getConversationById(id);
+  const data = setifyProperty(await channels.getConversationById(id), 'swarmNodes');
```

Comment:
```suggestion
  const rawData = await channels.getConversationById(id);
  const data = setifyProperty(rawData, 'swarmNodes');
```
@sachaaaaa @BeaudanBrown dunno if the above or the suggested seems better for readability?

Reply:
Yeah tru, how u feel about the map ones?